sql/src/main/java/io/crate/planner/optimizer/Optimizer.java

```diff
@@ -41,10 +42,15 @@
 
     private final List<Rule<?>> rules;
     private final Supplier<Version> minNodeVersionInCluster;
+    private final Functions functions;
 
-    public Optimizer(List<Rule<?>> rules, Supplier<Version> minNodeVersionInCluster) {
-        this.rules = rules;
+    public Optimizer(Functions functions,
+                     Supplier<Version> minNodeVersionInCluster,
+                     LoadedRules loadedRules,
+                     List<Class<? extends Rule<?>>> rulesToInclude) {
```

Comment:
Could the `loadedrules.rules(rulesToInclude)` call be done on the call-site? The call-site already has both components available. It would have the advantage that the `Optimizer` could be more easily unit-tested as the dependencies stay smaller.